### ISSUE TITLE
[Feature Request] Controller Mapping - Listed buttons ordered by category

### ISSUE THREAD

**Issue by pokeshark** [flyinghead/flycast#346]
Platform / OS / Hardware: Naomi / Win10 / PC

Could we please have a more consistent buttons order (by category) for items in both lists (Buttons and Analog Axes)?

An example of such items order:
- System Buttons (Start, Coin, Test..)
- Directions (Up, Down, Left, Right)
- Action Buttons (Button 1.. Button 8)
- Action Hotkeys (1+2+3, 4+5+6, ...)
- UI Actions (Menu, Exit, Fast Forward)
- Recording Actions (Quick Load, Quick Save, Record 1..3, Play 1..3, Switch Player)

For the user it would be easier to follow a list ordered by category.
Right now, the flow of mapping controls is pretty hard to follow.

This applies to both Arcade Controls and Dreamcast Controls mapping conventions. I used Arcade Controls since the mapping layout is more familiar to me.

Thank you for your time!

**_Screenshots_**
Current state:
![image](https://user-images.githubusercontent.com/10551675/132813150-f219a330-955a-4b88-b833-7c7fb8d89932.png)

Desired state:
Order of listed buttons re-arranged in consistent categories. Up to developer to decide on this order.
![image](https://user-images.githubusercontent.com/10551675/132816384-8491a9ac-1e3c-4c1a-8ae7-fedb34c36cd2.png)

**Comment by flyinghead** [comment]
This is implemented in the latest net-rollback build and in Flycast Dojo test build 9.

**Comment by pokeshark** [comment]
closing this then.. and thank you!
